aiverify-foundation/moonshot-data
# answercarefully test

## Description

add testing files for AnswerCarefullyVersion002_Test

## Motivation and Context

Testing the safety of responses in multiple languages ensures that LLMs remain inclusive, ethical, and reliable for a diverse global audience. It also helps identify and address biases, cultural differences, and technical gaps in training data, leading to better performance and user trust.

## Type of Change


## How to Test

I added four languages ​​of the AnswerCarefullyVersion002_Test dataset in the “datasets” folder, including English (jp-aisi-en.json), Chinese (jp-aisi-ch.json), Japanese (jp-aisi-jp.json), and Korean (jp-aisi-ko.json).

You can check the safety of answers from three models: Llama 3.1, 405B//Mixtral-8x22B Instruct v0.1//GDM Gemma 2, 27B judged by gpt4o.

Take the English (jp-aisi-en.json) for example:
Please run the command line:
run_recipe "my run jp-aisi-en dataset push to github together llama3.1 405b gpt4o " "['jp-aisi-en']" "['my-together-meta-llama-3-1-405b-instruct-connector']" -n 10 -r 1 -s " "
Expected testing result:
![image](https://github.com/user-attachments/assets/7ec53d70-9e80-4ef4-9e52-b6d431df0fd7)

run_recipe "my run jp-aisi-en dataset push to github together mixtral 822b instruct vgpt4o" "['jp-aisi-en']" "['my-together-mixtral-8x22b']" -n 10 -r 1 -s "  "
Expected testing result:
![image](https://github.com/user-attachments/assets/1f02230c-e055-4317-ae35-cd275a5f1dbf)

run_recipe "my run jp-aisi-en dataset push to github together gmd gemma 2 27b vgpt4o" "['jp-aisi-en']" "['my-together-gemma2-27b']" -n 10 -r 1 -s " "
Expected testing result:
![image](https://github.com/user-attachments/assets/c83e14c7-d3fe-4bef-abf4-7eff076cbc6a)

Just replacing “jp-aisi-en” in the command line with the names of other recipes is ok.
Those are the test results for a subset with only 10 examples. If you need to test on the full dataset, please remove “-n 10” from the command line.



## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
